src/Microsoft.Extensions.DependencyInjection/ServiceLookup/FactoryService.cs

```diff
@@ -24,6 +24,8 @@ public ServiceLifetime Lifetime
 
         public Type ServiceType => Descriptor.ServiceType;
 
+        public Type ImplementationType => Descriptor.ImplementationType;
```

Comment:
I think this line is confusing since `Descriptor.ImplementationType` is always null. Reading this in isolation, I would think that wasn't the case.